apps/src/templates/ShowCodeToggle.js

```diff
@@ -115,18 +117,33 @@ class DropletCodeToggle extends Component {
   toggle = () => {
     // are we trying to toggle from blocks to text (or the opposite)
     let fromBlocks = studioApp().currentlyUsingBlocks();
-
     let result;
+
+    let errorHandler = new JavaScriptModeErrorHandler();
+    let dropletErrors = findDropletParseErrors(
+      studioApp().editor,
+      (lineNumber, message) => errorHandler.outputError(message, lineNumber)
+    );
+
+    if (dropletErrors) {
+      studioApp().showToggleBlocksError();
+      return;
+    }
+
     try {
       result = studioApp().editor.toggleBlocks();
     } catch (err) {
       result = {error: err, nonDropletError: true};
     }
     if (result && result.error) {
+      // With the addition of the `findDropletParseErrors` above, this try/
```

Comment:
maybe add a date here? then we can more quickly tell if this code can be removed based on how long it's been since the comment was added